actions_test.go

```diff
@@ -0,0 +1,160 @@
+package opslevel_test
+
+import (
+	ol "github.com/opslevel/opslevel-go/v2023"
+	"github.com/rocktavious/autopilot/v2022"
+	"testing"
+)
+
+func TestWebhookActionHeadersMarshalling(t *testing.T) {
+	// Arrange
+	unstructured := "Accept=\"application/vnd.github+json\"\nAuthorization=\"Bearer XXX\"\nContent-Type=\"application/json\"\nHost=\"app.opslevel.com\"\nOrigin=\"api.opslevel.com\""
+	structured := map[string]string{
+		"Host":          "app.opslevel.com",
+		"Origin":        "api.opslevel.com",
+		"Content-Type":  "application/json",
+		"Accept":        "application/vnd.github+json",
+		"Authorization": "Bearer XXX",
+	}
+	// Act
+	marshalled := ol.CustomActionsToHeaders(unstructured)
+	unmarshalled := ol.CustomActionsFromHeaders(structured)
+	// Assert
+	autopilot.Equals(t, structured, marshalled)
+	autopilot.Equals(t, unstructured, unmarshalled)
+}
+
+func TestCreateWebhookAction(t *testing.T) {
+	//Arrange
+	request := `{"query": 
+		"mutation($input:CustomActionsWebhookActionCreateInput!){customActionsWebhookActionCreate(input: $input){webhookAction{{ template "custom_actions_request" }},errors{message,path}}}",
+		"variables":{"input":{"headers":"Content-Type=\"application/json\"","httpMethod":"POST","liquidTemplate":"{\"token\": \"XXX\", \"ref\":\"main\", \"action\": \"rollback\"}","name":"Deploy Rollback","webhookUrl":"https://gitlab.com/api/v4/projects/1/trigger/pipeline"}}
```

Comment:
while i'm still not 100% happy with this method for generating test request and responses its MUCH better and should reduce the number of fixture files and limit us to only having template files

Reply:
One thing we might want to consider soon is switching the go template delims `{{` and `}}` to `[[` and `]]` so as to not visually conflict with the GraphQL & Json syntax